src/box/sql/insert.c

```diff
@@ -537,6 +482,65 @@ sqlInsert(Parse * pParse,	/* Parser context */
 		goto insert_cleanup;
 	}
 
+	if (pSelect && useTempTable) {
+		/* Invoke the coroutine to extract information from the SELECT
+		 * and add it to a transient table srcTab. The code generated
+		 * here is from the 4th template:
+		 *
+		 *      B: open temp table
+		 *      L: yield X, goto M at EOF
+		 *         insert row from R..R+n into temp table
+		 *         goto L
+		 *      M: ...
+		 */
```

Comment:
I suggest moving this comment out of this `if`. If you choose to apply the diff described above, perhaps this comment will be merged with the comment mentioned above.

Reply:
I have moved the comment `Invoke the coroutine to extract ...` out of the `if`, but decided not to merge it with the comment about the usage of the `useTempTable` variable.

Reply:
This comment could be shortened a bit as there is no more `if`. But the current version is also good, I think.

Reply:
I think we can keep the comment "as is".